xarray/core/indexing.py

```diff
@@ -516,13 +516,36 @@ def get_duck_array(self):
         return self.array
 
 
+class IndexingAdapter:
+    """Marker class for indexing adapters.
+
+    These classes translate between Xarray's indexing semantics and the underlying array's
+    indexing semantics.
+    """
+
+    # FIXME: @abstractmethod here
+    def get_duck_array():
+        raise NotImplementedError
+
+    async def async_get_duck_array():
```

Comment:
80/20 on this. Alternatively, we special case IndexingAdapter inside `async_get_duck_array` which seems worse to me.